setup.py

```diff
@@ -47,6 +48,93 @@ def find_files(pattern, path=None, root="rez"):
     long_description = f.read()
 
 
+class BuildPyWithRezBinsPatch(build_py.build_py):
+
+    def run(self):
+        build_py.build_py.run(self)
+        self.patch_rez_binaries()
+        self.copy_completion_scripts()
+
+    def _append(self, data_files):
+        """Append `data_files` into `distribution.data_files`
+
+        Just like how additional files be assigned with setup(data_files=[..]),
+        but for those extra files that can only be created in build time, here
+        is the second chance.
+
+        The `data_files` specifies a sequence of (directory, files) pairs in
+        the following way:
+
+            setup(...,
+                data_files=[('config', ['foo/cfg/data.cfg'])],
+            )
+
+        Each (directory, files) pair in the sequence specifies the installation
+        directory and the files to install there.
+
+        So in the example above, the file `data.cfg` will be installed to
+        `config/data.cfg`.
+
+        IMPORTANT:
+        The directory MUST be a relative path. It is interpreted relative to
+        the installation prefix (Python’s sys.prefix for system installations;
+        site.USER_BASE for user installations).
+
+        @param data_files: a sequence of (directory, files) pairs
+        @return:
+        """
+        # will be picked up by `distutils.command.install_data`
+        if self.distribution.data_files is None:
+            self.distribution.data_files = data_files
+        else:
+            self.distribution.data_files += data_files
+
+    def patch_rez_binaries(self):
+        from rez.vendor.distlib.scripts import ScriptMaker
+
+        self.announce("Generating rez bin tools...", level=3)
+
+        # Create additional build dir for binaries, so they won't be handled
+        # as regular builds under "build/lib".
+        build_path = os.path.join("build", "rez_bins")
+        self.mkpath(build_path)
+
+        # Make binaries, referenced from rez's install.py
+        maker = ScriptMaker(
+            source_dir=None,
+            target_dir=build_path
+        )
+        maker.executable = sys.executable
+        rel_rez_bin_paths = maker.make_multiple(
+            specifications=get_specifications().values(),
+            options=dict(interpreter_args=["-E"])
+        )
+
+        # Add validation file as production rez install
+        # Do not remove - rez uses this!
+        validation_file = os.path.join(build_path, ".rez_production_install")
+        with open(validation_file, "w") as vfn:
+            vfn.write(_rez_version)
+        rel_rez_bin_paths.append(validation_file)
+
+        # Compute relative install path, to work with wheel.
+        # Install path, e.g. "bin/rez" or "scripts/rez" on Windows.
+        abs_rez_dir = os.path.join(os.path.dirname(sys.executable), "rez")
+        rel_rez_dir = os.path.relpath(abs_rez_dir, sys.prefix)
```

Comment:
This may get unexpected result if `pip install` with `--prefix` or `--user`, the relative path should be calculated in `install` step.